src/main/java/org/zalando/nakadi/repository/kafka/KafkaFactory.java

```diff
@@ -8,22 +15,73 @@
 import org.springframework.context.annotation.Profile;
 import org.springframework.stereotype.Component;
 
-import java.util.List;
-
 @Component
 @Profile("!test")
 public class KafkaFactory {
+
     private final KafkaLocationManager kafkaLocationManager;
-    private final KafkaProducer<String, String> kafkaProducer;
+    @Nullable
+    private Producer<String, String> activeProducer;
+    private final Map<Producer<String, String>, AtomicInteger> useCount = new ConcurrentHashMap<>();
```

Comment:
No, because of several reasons - Producers are taken and released under read lock, which means, that they are accessed from multiple threads simultaneously. That means, that it is impossible to use useCount variable without additional locking (that is why AtomicInteger)
`put` call to useCount map by itself is made with writeLock, while getting data is made without lock at all, and removing from this map is made with readLock. That means, that map by itself should be somehow synchronized (that is why `ConcurrentHashMap`)